Fix #227

spicychickensauce opened the pull request:
I took your test as a starting point and implemented a possible fix for https://github.com/germsvel/phoenix_test/issues/227

I made the minimal changes needed to fix this issue, however I think a proper fix would affect a lot more code.
Because of that I kept this PR in a draft state, as I don't want to do many invasive changes.

Essentially everywhere a struct like `Button` or e.g. `Field` is used, the constructed selector needs to respect the current `within`. So this bug is almost certainly also present outside of `click_button`.

Instead of what I did in my fix, a better approach would probably be to pass around the current session into `Button.find!` and similar such that the correct selector can be constructed then.

germsvel commented:
@spicychickensauce just took a look at this. Thanks for opening it! 

It looks mostly right to me. But, any chance you could move the `scope_selector` calls right above the code blocks where we call `render_click`? I'd like to keep the scoping to a minimum (in terms of surface area in the code).

spicychickensauce commented:
@germsvel Alright, I changed it as you suggested :+1: 
There were some places where I removed the `scope_selector` call completely (`upload` and `fill_in`), as the `field.selector` wasn't actually used.
Those probably do have this bug, but maybe better to address it once someone actually runs into it, as they work a bit differently

germsvel commented:
Thanks @spicychickensauce!
